examples/text_style_transfer/main.py

```diff
@@ -0,0 +1,210 @@
+# Copyright 2018 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Text style transfer
+
+This is a simplified implementation of:
+
+Toward Controlled Generation of Text, ICML2017
+Zhiting Hu, Zichao Yang, Xiaodan Liang, Ruslan Salakhutdinov, Eric Xing
+
+Download the data with the cmd:
+
+$ python prepare_data.py
+
+Train the model with the cmd:
+
+$ python main.py --config config
+"""
+
+# pylint: disable=invalid-name, too-many-locals, too-many-arguments, no-member
+
+import os
+import importlib
+import argparse
+import numpy as np
+import torch
+
+import texar.torch as tx
+
+from ctrl_gen_model import CtrlGenModel
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument('--config', default='config', help="The config to use.")
+
+args = parser.parse_args()
+
+config = importlib.import_module(args.config)
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+def _main():
+    # Data
+    train_data = tx.data.MultiAlignedData(hparams=config.train_data,
+                                          device=device)
+    val_data = tx.data.MultiAlignedData(hparams=config.val_data,
+                                        device=device)
+    test_data = tx.data.MultiAlignedData(hparams=config.test_data,
+                                         device=device)
+    vocab = train_data.vocab(0)
+
+    # Each training batch is used twice: once for updating the generator and
+    # once for updating the discriminator. Feedable data iterator is used for
+    # such case.
+    iterator = tx.data.DataIterator(
+        {'train_g': train_data, 'train_d': train_data,
+         'val': val_data, 'test': test_data})
+
+    # Model
+    gamma_ = 1.
+    lambda_g_ = 0.
+
+    # Model
+    model = CtrlGenModel(vocab, hparams=config.model)
+    model.to(device)
+
+    # create optimizers
+    train_op_d = tx.core.get_optimizer(
+        params=model.d_vars,
+        hparams=config.model['opt']
+    )
+
+    train_op_g = tx.core.get_optimizer(
+        params=model.g_vars,
+        hparams=config.model['opt']
+    )
+
+    train_op_g_ae = tx.core.get_optimizer(
+        params=model.g_vars,
+        hparams=config.model['opt']
+    )
+
+    def _train_epoch(gamma_, lambda_g_, epoch, verbose=True):
```

Comment:
In `texar-tf`, both `train_d` and `train_g` are used in the `_train_epoch`, but we only use `train_g` in this function. Did I miss something important?